appcelerator-archive/amp
# Amp cluster management

Related to #512 

For the ones granted, here is a video to explain more in details the PR features
https://drive.google.com/a/axway.com/file/d/0B3HdJx5HHXGcUzR1ZVRSMGVURVk/view?usp=sharing

see also the discussion on the topic "Short demo on amp cluster features" on Mobilize

to test locally:
`make install`
`./shrink.sh local`
`ampadm pf pull --local -v`
`ampadm pf start --local -v`

`ampadm pf monitor` (on another console)
`ampadm node ls`
`ampadm node count`
`ampadm pf stop`

to test on Amazon amp cluster after having used the Nicolas's amp-bootstrap installation:
On the manager execute:
`ampadm cluster start --tag cluster`
(it use a predefined amp:cluster image for PR need, but after merge and image appelerator/amp:latest built and pulled no need of the option "--tag cluster")
That's all on manager.

on your laptop client machine execute:
`ampadm --server [managerAddress]:31315 node ls`  (to see the nodes status)
`ampadm --server [managerAddress]:31315 node purge --container --image` (if needed)
`ampadm --server [managerAddress]:31315 pf pull -v` (all amp images are pulled on all nodes)
`ampadm --server [managerAddress]:31315 pf start -v`
`ampadm --server [managerAddress]:31315 pf monitor` (on another console)

[managerAdresse] can be an aws load-balancer address

For now the previous amp pf commands stay available, waiting for decision about amp/amp-cluster cli integration or not.

-ampadm --server [managerAddress]:31315 pf pull -v
